### PR TITLE
Guard out-of-range shift counts in integer opcodes

### PR DESCRIPTION
**Pull request by uwezkhan**
The integer shift opcodes hand the per-element shift count straight to C++ `<<`/`>>`. When `a<<b` or `a>>b` runs with a count that is negative or at least the operand width (32 for int, 64 for long long) the shift is undefined behavior. On arm64 `evaluate("a<<b")" with b=100 returns 80 because the hardware masks the count, while NumPy returns 0, and a UBSAN build traps right at the shift. Both int and long long, left and right, hit this.

After the change an out-of-range left shift yields 0 and an out-of-range right shift clamps the count to width-1 so the sign bit fills, which is the result NumPy gives. The guard sits in the opcode next to the existing div/mod guards because the count is only known per element at run time, so a caller-side check could not cover array shift counts. Tradeoff is one extra unsigned compare per element; in-range counts keep the same value and the branch is well predicted.

### CODE REVIEW

**Comment by uwezkhan** [comment]
gentle ping

**Comment by uwezkhan** [comment]
Rebased onto master. The only red jobs were the free-threaded 3.13 wheel builds, and those were failing while building numpy from source rather than on this change; master already dropped them in bd99ef3, so they no longer run. Also extended the test to hit the exact width boundary (32 for i4, 64 for i8) that Copilot flagged, not just the far out-of-range counts.

File: numexpr/tests/test_numexpr.py [hunk 1]
```diff
@@ -457,6 +457,17 @@ def test_right_shift(self):
...
+    def test_shift_out_of_range(self):
+        # Shift counts that are negative or >= the operand width are
+        # undefined behavior in C. Match NumPy, which treats them as a
+        # full shift (0 for <<, sign fill for >>).
+        for dtype in ('i4', 'i8'):
+            x = array([5, -5, 0], dtype=dtype)
+            for count in (-1, 64, 200):
+                y = array([count] * len(x), dtype=dtype)
+                assert_array_equal(evaluate("x << y"), x << y)
+                assert_array_equal(evaluate("x >> y"), x >> y)
```
[the remark this answers is not in the document]
Comment (uwezkhan) [reply numexpr/tests/test_numexpr.py:498]: Good point. Derived the width from the dtype and added count == width and width+1, so the >= width edge is covered for both i4 and i8.

**Comment by FrancescAlted** [comment]
Thanks @uwezkhan !
